src/timeseriesflattener/flattened_dataset.py

```diff
@@ -27,69 +27,105 @@ def add_outcome(
         self,
         outcome_df: DataFrame,
         lookahead_days: float,
-        resolve_multiple: str,
-        fallback: List[str],
-        values_colname: str = "values",
+        resolve_multiple: Callable,
+        fallback: float,
+        source_values_colname: str = "val",
         new_col_name: str = None,
     ):
         """Adds an outcome-column to the dataset
 
         Args:
             outcome_df (DataFrame): Cols: dw_ek_borger, datotid, value if relevant.
             lookahead_days (float): How far ahead to look for an outcome in days. If none found, use fallback.
-            resolve_multiple (str): What to do with more than one value within the lookahead.
-                Suggestions: earliest, latest, mean, max, min.
-            fallback (List[str]): What to do if no value within the lookahead.
-                Suggestions: latest, mean_of_patient, mean_of_population, hardcode (qualified guess)
-            values_colname (str): Colname for outcome values in outcome_df
+            resolve_multiple (Callable): How to handle multiple values within the lookahead window. Takes a a function that takes a list as an argument and returns a float.
+            fallback (float): What to do if no value within the lookahead.
+            source_values_colname (str): Colname for outcome values in outcome_df. Defaults to "val".
             new_col_name (str): Name to use for new col. Automatically generated as '{new_col_name}_within_{lookahead_days}_days'.
-                Defaults to using values_colname.
         """
 
-        outcome_dict = self._events_to_dict_by_patient(
-            df=outcome_df,
-            values_colname=values_colname,
-        )
-
-        new_col = self.df_prediction_times.apply(
-            lambda row: self._flatten_events_for_prediction_time(
-                direction="ahead",
-                prediction_timestamp=row[self.timestamp_colname],
-                val_dict=outcome_dict,
-                interval_days=lookahead_days,
-                id=row[self.id_colname],
-                resolve_multiple=resolve_multiple,
-                fallback=fallback,
-            ),
-            axis=1,
+        self._add_col(
+            values_df=outcome_df,
+            direction="ahead",
+            interval_days=lookahead_days,
+            resolve_multiple=resolve_multiple,
+            fallback=fallback,
+            new_col_name=new_col_name,
+            source_values_colname=source_values_colname,
```

Comment:
I have no idea what this does - it adds a new column - a comment could probably do.

Reply:
Renamed the method. Since `add_predictor` and `add_outcome` share such a large amount of code, I decided to have an internal method with a direction argument.

I still think having to user-facing methods is a good idea, to make the code easier to interpret.

Reply:
I do to, but the `_add_col` could basically be any kind of function even if it is internal it should be possible to interpret what the function does.